bubblegum/program/src/lib.rs

```diff
@@ -603,9 +726,12 @@ pub mod bubblegum {
         let merkle_slab = &ctx.accounts.merkle_slab;
 
         // The mint authority must sign if it is not equal to the tree authority
-        if mint_authority.key() != ctx.accounts.authority.key() {
+        let signer = if mint_authority.key() != ctx.accounts.authority.key() {
```

Comment:
This particular concept needs close look and review and please feel free to ping me with offline discussion about if I'm doing this wrong.

Basically if the mint_authority is a signer it can be used to verify a creator, but not the tree authority if the tree authority is not a signer.  I'm not 100% sure I fully understand if that is correct behavior.